include/nbl/builtin/hlsl/emulated/float64_t.hlsl

```diff
@@ -0,0 +1,623 @@
+                }
+
+                mantissa &= ieee754::traits<float64_t>::mantissaMask;
+
+                uint64_t output = emulated_float64_t_impl::assembleFloat64(sign, uint64_t(exp) << ieee754::traits<float64_t>::mantissaBitCnt, mantissa);
+                output = emulated_float64_t_impl::flushDenormToZero(output);
+                return bit_cast<this_t>(output);
+            }
+            else
+            {
+                //static_assert(false, "not implemented yet");
+                return bit_cast<this_t>(0xdeadbeefbadcaffeull);
+            }
+        }
+
+        emulated_float64_t operator/(const float rhs) NBL_CONST_MEMBER_FUNC
+        {
+            return bit_cast<this_t>(data) / create(rhs);
+        }
+
+        // relational operators
+        bool operator==(this_t rhs) NBL_CONST_MEMBER_FUNC
+        {
+            if (!FastMath)
+            {
+                if (tgmath::isNaN<uint64_t>(data) || tgmath::isNaN<uint64_t>(rhs.data))
+                    return false;
+                if (emulated_float64_t_impl::areBothZero(data, rhs.data))
+                    return true;
+            }
+
+            return data == rhs.data;
+        }
+        bool operator!=(emulated_float64_t rhs) NBL_CONST_MEMBER_FUNC
+        {
+            if (!FastMath && (tgmath::isNaN<uint64_t>(data) || tgmath::isNaN<uint64_t>(rhs.data)))
+                return false;
+
+            return !(bit_cast<this_t>(data) == rhs);
+        }
+        bool operator<(emulated_float64_t rhs) NBL_CONST_MEMBER_FUNC
+        {
+            if (!FastMath)
+            {
+                if (tgmath::isNaN<uint64_t>(data) || tgmath::isNaN<uint64_t>(rhs.data))
+                    return false;
+                if (emulated_float64_t_impl::areBothInfinity(data, rhs.data))
+                {
+                    if(ieee754::extractSignPreserveBitPattern(data) == ieee754::extractSignPreserveBitPattern(rhs.data))
+                        return false;
+                }
+                if (emulated_float64_t_impl::areBothZero(data, rhs.data))
+                    return false;
+            }
+
+            const uint64_t lhsSign = ieee754::extractSign(data);
+            const uint64_t rhsSign = ieee754::extractSign(rhs.data);
+
+            // flip bits of negative numbers and flip signs of all numbers
+            uint64_t lhsFlipped = data ^ ((0x7FFFFFFFFFFFFFFFull * lhsSign) | ieee754::traits<float64_t>::signMask);
+            uint64_t rhsFlipped = rhs.data ^ ((0x7FFFFFFFFFFFFFFFull * rhsSign) | ieee754::traits<float64_t>::signMask);
+
+            return lhsFlipped < rhsFlipped;
+        }
+        bool operator>(emulated_float64_t rhs) NBL_CONST_MEMBER_FUNC
+        {
+            if (!FastMath)
+            {
+                if (tgmath::isNaN<uint64_t>(data) || tgmath::isNaN<uint64_t>(rhs.data))
+                    return false;
+                if (emulated_float64_t_impl::areBothInfinity(data, rhs.data))
+                    return false;
```

Comment:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/696#discussion_r1817860645

See if you can generate the `operator<` and `>` from an `__private_method` template or something so they're both perfectly in sync and DRY